File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        TreeMap<Integer,Integer> numberAndCount = new TreeMap<>();
        for(int num:nums)
        {
            if(numberAndCount.containsKey(num))
            {
                numberAndCount.put(num,numberAndCount.get(num)+1);
            }
            else 
            {
                numberAndCount.put(num,1);
            }
        }
        int i = 0;
        for(int key:numberAndCount.keySet())
        {
            nums[i] = key;
            i++;
        }
      
        return numberAndCount.size();
    }
}

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int ind = 0;
        int n = nums.size();
        for(int i=0;i<n;i++)
        {
            if(ind!=0)
            {
                if(nums[ind-1]==nums[i]) continue;
                nums[ind] = nums[i];
            }
            ind++;
        }
        return ind;
    }
};